scrunch/tests/test_datasets.py

```diff
@@ -163,9 +163,9 @@ def _get(*args):
 
 class TestDatasets(TestDatasetBase, TestCase):
 
-    def test_edit_Dataset(self):
+    def test_edit_dataset(self):
         ds_mock = self._dataset_mock()
-        ds = Dataset(ds_mock)
+        ds = MutableDataset(ds_mock)
```

Comment:
When the methods mentioned in the comments above are moved to BaseDataset, here you will be able to use StreamingDataset and only use MutableDataset in the tests where they need that mutable method.